Subject: rt/rcutree: Move misplaced prototype
From: Ingo Molnar <mingo@elte.hu>
Date: Wed Dec 14 12:51:28 CET 2011

Fix this warning on x86 defconfig:

  kernel/rcutree.h:433:13: warning: ‘rcu_preempt_qs’ declared ‘static’ but never defined [-Wunused-function]

The #ifdefs and prototypes here are a maze, move it closer to the
usage site that needs it.

Signed-off-by: Ingo Molnar <mingo@elte.hu>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
---
 kernel/rcutree.c |    2 ++
 kernel/rcutree.h |    1 -
 2 files changed, 2 insertions(+), 1 deletion(-)

Index: linux-3.4/kernel/rcutree.c
===================================================================
--- linux-3.4.orig/kernel/rcutree.c
+++ linux-3.4/kernel/rcutree.c
@@ -173,6 +173,8 @@ void rcu_sched_qs(int cpu)
 }
 
 #ifdef CONFIG_PREEMPT_RT_FULL
+static void rcu_preempt_qs(int cpu);
+
 void rcu_bh_qs(int cpu)
 {
 	rcu_preempt_qs(cpu);
Index: linux-3.4/kernel/rcutree.h
===================================================================
--- linux-3.4.orig/kernel/rcutree.h
+++ linux-3.4/kernel/rcutree.h
@@ -423,7 +423,6 @@ DECLARE_PER_CPU(char, rcu_cpu_has_work);
 /* Forward declarations for rcutree_plugin.h */
 static void rcu_bootup_announce(void);
 long rcu_batches_completed(void);
-static void rcu_preempt_qs(int cpu);
 static void rcu_preempt_note_context_switch(int cpu);
 static int rcu_preempt_blocked_readers_cgp(struct rcu_node *rnp);
 #ifdef CONFIG_HOTPLUG_CPU
